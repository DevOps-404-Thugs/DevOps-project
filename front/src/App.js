import React, { Suspense } from 'react';
import { HashRouter, Route, Switch } from "react-router-dom";

import MainPage from './Components/MainPage/MainPage';
import Detail from './Components/Detail/Detail';
import Upload from './Components/Upload/Upload';
import Login from './Components/Login/Login';
import Register from './Components/Register/Register';


function App() {
  return (
    <div className="App">
      <HashRouter>
        <Switch>
          <Route exact path='/' component ={MainPage}/>
          <Route exact path='/detail' component ={Detail}/>
          {/* <Route exact path='/detail/:id' component ={Detail}/> */}
          <Route exact path='/upload' component ={Upload}/>
          <Route exact path='/login' component ={Login}/>
          <Route exact path='/register' component ={Register}/>
        </Switch>
      </HashRouter>
      
    </div>
  );
}

export default App;
