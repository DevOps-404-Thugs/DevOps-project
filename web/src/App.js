import React from 'react';
import ReactDOM from 'react-dom';
import logo from './logo.svg';
import './App.css';
import Detail from './Detail';

function DetailLink() {
  function handleClick(e) {
    ReactDOM.render(
      <React.StrictMode>
        <Detail />
      </React.StrictMode>,
      document.getElementById('root')
    );
  }

  return (
    <button onClick={handleClick}>
      Detail
    </button>
  );
}

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <DetailLink />
      </header>
      
    </div>
  );
}

export default App;
