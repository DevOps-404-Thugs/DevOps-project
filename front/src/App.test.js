import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('Description generates correct page', () => {
  const component = renderer.create(
      <App />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
