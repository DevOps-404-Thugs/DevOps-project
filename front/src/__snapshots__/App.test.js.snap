// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description generates correct page 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/ihomie"
    >
      iHomie
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/ihomie"
          href="/ihomie"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/ihomie#/upload"
          href="/ihomie#/upload"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Upload
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/ihomie#/register"
          href="/ihomie#/register"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Register
        </a>
      </div>
      <form
        className="form-inline"
      >
        <button
          className="btn btn-outline-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </form>
    </div>
  </nav>
  <div
    style={
      Object {
        "margin": "3rem auto",
        "width": "65%",
      }
    }
  >
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h2>
          Find Your Future Home in iHomie 
      </h2>
    </div>
    <div>
      <div
        className="ant-row"
        style={
          Object {
            "marginBottom": 8,
            "marginLeft": -8,
            "marginRight": -8,
            "marginTop": -8,
          }
        }
      />
    </div>
    <br />
    <br />
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <a
        className="link"
        href="#/upload"
        onClick={[Function]}
      >
        Upload New House
      </a>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <a
        className="link"
        href="#/account_update"
        onClick={[Function]}
        style={
          Object {
            "color": "orange",
          }
        }
      >
        Update Account Information
      </a>
    </div>
  </div>
</div>
`;
